codec/types/any.go

```diff
@@ -98,6 +99,23 @@ func UnsafePackAny(x interface{}) *Any {
 	return &Any{cachedValue: x}
 }
 
+// PackAny is a checked and safe version of UnsafePackAny. It assures that
+// `x` implements the proto.Message interface and uses it to serialize `x`.
+// TODO: should be moved away: https://github.com/cosmos/cosmos-sdk/issues/7479
+func PackAny(x interface{}) (*Any, error) {
```

Comment:
```suggestion
func LegacyPackAny(x interface{}) (*Any, error) {
```

The name should clearly indicate that this is only for legacy code. I would even add a deprecated comment.

Reply:
I've market this function as deprecated.